datavault-webapp/src/main/webapp/WEB-INF/freemarker/admin/roles/index.ftl

```diff
@@ -0,0 +1,345 @@
+
+            $.ajax({
+                method: "GET",
+                url: '${springMacroRequestContext.getContextPath()}/admin/roles/'+roleid,
+                   success: function (data) {
+
+                    $("model_msg").innerText = "";
+
+                    $("#modalnewedit").modal("show");
+
+                    $("#edit_roleid").val(data.role.id)
+                    $("#edit_name").val(data.role.name);
+                    $("#edit_description").val(data.role.description);
+                    $("#edit_type").val(data.role.type);
+
+
+                    $("#edit_vaultpermissions").empty();
+                   data.role.permissions.forEach(function(p){
+                           setPermissionsHtml(p, true, "edit_vaultpermissions");
+                       });
+
+                    data.permissions.forEach(function(p){
+                       setPermissionsHtml(p, false, "edit_vaultpermissions");
+                    });
+
+                },
+                error: function(xhr, ajaxOptions, thrownError) {
+                    $("#modalMsg").show();
+                    $("#modalMsg").html("<strong>Error Fetching Role: </strong> " + thrownError);
+                }
+            });
+        }
+
+        $("#modalSave").click(function(){
+            $("#modalMsg").hide();
+
+            if(!$("#edit_name").val()){
+                $("#modalMsg").show();
+                $("#modalMsg").html("Need role name to save");
+                return;
+            }
+
+            if( $("[name='permisionmodel']").filter(':checked').length == 0)
+            {
+                $("#modalMsg").show();
+                $("#modalMsg").html("Need at least One permission selected");
+                return;
+            }
+
+            saveform();
+        });
+
+        function saveform() {
+            var formdata = {
+                id: $("#edit_roleid").val(),
+                name: $("#edit_name").val(),
+                description: $("#edit_description").val(),
+                type: $("#edit_type").val(),
+                permissions: []
+
+            };
+
+            $("[name='permisionmodel']").filter(':checked').each(function(i){
+               formdata.permissions.push(this.id);
+            });
+
+            $.ajax({
+                url: '${springMacroRequestContext.getContextPath()}/admin/roles/save',
+                type: 'POST',
+                data: formdata,
+                success: function() {
+                    //Redirect
+                    window.location="${springMacroRequestContext.getContextPath()}/admin/roles";
+                },
+                error: function(xhr, ajaxOptions, thrownError) {
+                  $("#modalMsg").show();
+                  $("#modalMsg").html("<strong>Error: </strong> " + thrownError);
+                }
+            });
+        }
+
+
+
+
+
+        function setPermissionsHtml(permission, checked, target)
+        {
+            var outdiv = document.createElement("div");
```

Comment:
Might as well use jQuery if you've got it:

```js
var el = $('<div/>').addClass('col-xs-6 form-group form-inline');
// ...
```

Reply:
I disagree here,

Reply:
...

For any particular reason?

Reply:
document.createElement is safer than injection of a string html into page (as browser creates html required based on its own standards).  And if you look at the code setPermissionsHtml function it inserts a single element to a target (used in multiple places),  using the method you suggest then I assume you would use .innerHMTL which would replace the target value so you would need to concatenate multiple html fragments then inject.  For me I'll take code over html string concatenation any day. You could argue the innerHTML is faster but for maintainability I'd prefer createElement.

Reply:
I never mentioned anything about string concatenation or setting `innerHTML` properties. The way `jQuery()` works is by scanning the input string for an XML tag and creating an element based on that tag, using the exact same APIs as you have.

> as browser creates html required based on its own standards

There's no observable difference between the result of `createElement('div')` across all 3 major browsers, so I'm not sure what you mean. The only real difference is that some browsers will (incorrectly) accept `createElement('<div>')`.

> And if you look at the code setPermissionsHtml function it inserts a single element to a target (used in multiple places

Right, `el.appendTo(target)`.

The point of this comment is that you were using jQuery literally everywhere but here and there doesn't to be any reason for that.